railties/lib/rails/test_unit/testing.rake

```diff
@@ -30,35 +26,29 @@ namespace :test do
 
   ["models", "helpers", "channels", "controllers", "mailers", "integration", "jobs", "mailboxes"].each do |name|
     task name => "test:prepare" do
-      $: << "test"
-      Rails::TestUnit::Runner.rake_run(["test/#{name}"])
+      sh("rails", "test", "test/#{name}", *Rails::TestUnit::Runner.rake_test_opts, verbose: false)
     end
   end
 
   desc "Runs all tests, including system tests"
   task all: "test:prepare" do
-    $: << "test"
-    Rails::TestUnit::Runner.rake_run(["test/**/*_test.rb"])
+    sh("rails", "test", "test/**/*_test.rb", *Rails::TestUnit::Runner.rake_test_opts, verbose: false)
   end
 
   task generators: "test:prepare" do
-    $: << "test"
-    Rails::TestUnit::Runner.rake_run(["test/lib/generators"])
+    sh("rails", "test", "test/lib/generators", *Rails::TestUnit::Runner.rake_test_opts, verbose: false)
   end
 
   task units: "test:prepare" do
-    $: << "test"
-    Rails::TestUnit::Runner.rake_run(["test/models", "test/helpers", "test/unit"])
+    sh("rails", "test", "test/models", "test/helpers", "test/unit", *Rails::TestUnit::Runner.rake_test_opts, verbose: false)
   end
 
   task functionals: "test:prepare" do
-    $: << "test"
-    Rails::TestUnit::Runner.rake_run(["test/controllers", "test/mailers", "test/functional"])
+    sh("rails", "test", "test/controllers", "test/mailers", "test/functional", *Rails::TestUnit::Runner.rake_test_opts, verbose: false)
   end
 
   desc "Run system tests only"
   task system: "test:prepare" do
-    $: << "test"
-    Rails::TestUnit::Runner.rake_run(["test/system"])
+    sh("rails", "test", "test/system",  *Rails::TestUnit::Runner.rake_test_opts, verbose: false)
```

Comment:
What does `verbose: false` do in these `sh` calls? I mean, what would be printed without it that's undesirable?

Reply:
It makes rake not output the command before running it, since `rake test` wasn't before I wanted to not change that. But using `system` would also not output the command so I'm going to go with that.